packages/block-editor/src/components/global-styles/filters-panel.js

```diff
@@ -0,0 +1,157 @@
+	const defaultPresets =
+		settings?.color?.[ presetSetting ]?.default || EMPTY_ARRAY;
+	return useMemo(
+		() => [
+			...userPresets,
+			...themePresets,
+			...( disableDefault ? EMPTY_ARRAY : defaultPresets ),
+		],
+		[ disableDefault, userPresets, themePresets, defaultPresets ]
+	);
+}
+
+export function useHasFiltersPanel( settings ) {
+	const hasDuotone = useHasDuotoneControl( settings );
+
+	return hasDuotone;
+}
+
+function useHasDuotoneControl( settings ) {
+	return settings.color.customDuotone || settings.color.defaultDuotone;
+}
+
+function FiltersToolsPanel( {
+	resetAllFilter,
+	onChange,
+	value,
+	panelId,
+	children,
+} ) {
+	const resetAll = () => {
+		const updatedValue = resetAllFilter( value );
+		onChange( updatedValue );
+	};
+
+	return (
+		<ToolsPanel
+			label={ __( 'Filters' ) }
+			resetAll={ resetAll }
+			panelId={ panelId }
+		>
+			{ children }
+		</ToolsPanel>
+	);
+}
+
+const DEFAULT_CONTROLS = {
+	duotone: true,
+};
+
+export default function FiltersPanel( {
+	as: Wrapper = FiltersToolsPanel,
+	value,
+	onChange,
+	inheritedValue = value,
+	settings,
+	panelId,
+	defaultControls = DEFAULT_CONTROLS,
+} ) {
+	const decodeValue = ( rawValue ) =>
+		getValueFromVariable( { settings }, '', rawValue );
+
+	// Duotone
+	const hasDuotoneEnabled = useHasDuotoneControl( settings );
+	const duotonePalette = useMultiOriginColorPresets( settings, {
+		presetSetting: 'duotone',
+		defaultSetting: 'defaultDuotone',
+	} );
+	const colorPalette = useMultiOriginColorPresets( settings, {
+		presetSetting: 'palette',
+		defaultSetting: 'defaultPalette',
+	} );
+	const duotone = decodeValue( inheritedValue?.filter?.duotone );
+	const setDuotone = ( newValue ) => {
+		const duotonePreset = duotonePalette.find( ( { colors } ) => {
+			return colors === newValue;
+		} );
+		const settedValue = duotonePreset
+			? `var:preset|duotone|${ duotonePreset.slug }`
+			: newValue;
+		onChange( immutableSet( value, [ 'filter', 'duotone' ], settedValue ) );
+	};
+	const hasDuotone = () => !! value?.filter?.duotone;
+	const resetDuotone = () => setDuotone( undefined );
+
+	const resetAllFilter = useCallback( ( previousValue ) => {
+		return {
+			...previousValue,
+			filter: {
+				...previousValue.filter,
+				duotone: undefined,
+			},
+		};
+	}, [] );
+
+	return (
+		<Wrapper
+			resetAllFilter={ resetAllFilter }
+			value={ value }
+			onChange={ onChange }
+			panelId={ panelId }
+		>
+			{ hasDuotoneEnabled && (
+				<ToolsPanelItem
+					label={ __( 'Duotone' ) }
+					hasValue={ hasDuotone }
+					onDeselect={ resetDuotone }
+					isShownByDefault={ defaultControls.duotone }
+					panelId={ panelId }
+				>
+					<VStack>
+						<p>
+							{ __(
+								'Create a two-tone color effect without losing your original image.'
+							) }
+						</p>
+						<DuotonePicker
+							colorPalette={ colorPalette }
+							duotonePalette={ duotonePalette }
+							disableCustomColors={ true }
+							disableCustomDuotone={ true }
```

Comment:
These flags are automatically disabled right now because I mapped the behavior in global styles. We should definitely unify here with block level and allow custom colors and custom duotone according to the "settings" object. cc @MaggieCabrera

Reply:
When custom and default are disabled, the filter section is still there, but there are no options inside, which makes sense when you look at the code. 

<img width="285" alt="Screenshot 2023-04-04 at 13 54 02" src="https://user-images.githubusercontent.com/3593343/229783658-1d4c86ff-04c4-437b-92d9-c40e95d9e8bd.png">

In the future, we want third parties to add other custom filters here, so this behavior makes sense, but until that is a possibility, we would need to follow up and completely hide the filters section if duotone is disabled, is that a correct assumption?

Reply:
> When custom and default are disabled

My goal was to hide the panel entirely, maybe I missed something, I'll check

Reply:
I think this should be fixed now.